### ISSUE TITLE
Multiple errors in compiling and building the game

### ISSUE THREAD

**Issue by sudg6** [NetHack/NetHack#847]
Following the instructions provided by the readme files and install.unix accompanied by the repo. leads to compiling errors.
The compiler I'm using gcc (Ubuntu 11.2.0-19ubuntu1) 11.2.0. The first lines of `make all` output reads,
`( cd src ; make )
make[1]: Entering directory '/home/myusername/Downloads/NetHack-NetHack-3.6.6_Released/src'
cc -g -O -I../include -DNOTPARMDECL -DDLB -DCOMPRESS=\"/bin/gzip\" -DCOMPRESS_EXTENSION=\".gz\" -DSYSCF -DSYSCF_FILE=\"/root/nh/install/games/lib/nethackdir/sysconf\" -DSECURE -DTIMED_DELAY -DHACKDIR=\"/root/nh/install/games/lib/nethackdir\" -DDUMPLOG -DCONFIG_ERROR_SECURE=FALSE -DCURSES_GRAPHICS -c monst.c
In file included from /usr/include/features.h:486,
                 from /usr/include/x86_64-linux-gnu/bits/libc-header-start.h:33,
                 from /usr/include/stdio.h:27,
                 from ../include/global.h:9,
                 from ../include/config.h:581,
                 from monst.c:6:
/usr/include/x86_64-linux-gnu/sys/cdefs.h:705:2: error: #endif without #if
  705 | #endif   /* sys/cdefs.h */
      |  ^~~~~
In file included from /usr/include/features.h:486,
                 from /usr/include/x86_64-linux-gnu/bits/libc-header-start.h:33,
                 from /usr/include/stdio.h:27,
                 from ../include/global.h:9,
                 from ../include/config.h:581,
                 from monst.c:6:
/usr/include/stdio.h:189:46: error: expected ‘,’ or ‘;’ before ‘__attribute_warn_unused_result__’
  189 |   __attribute_malloc__ __attr_dealloc_fclose __wur;
      |                                              ^~~~~
/usr/include/stdio.h:201:47: error: expected ‘,’ or ‘;’ before ‘__attribute_warn_unused_result__’
  201 |    __attribute_malloc__ __attr_dealloc_fclose __wur;
      |                                               ^~~~~
/usr/include/stdio.h:205:46: error: expected ‘,’ or ‘;’ before ‘__attribute_warn_unused_result__’
  205 | extern char *tmpnam (char[L_tmpnam]) __THROW __wur;
      |                                              ^~~~~
/usr/include/stdio.h:210:52: error: expected ‘,’ or ‘;’ before ‘__attribute_warn_unused_result__’
  210 | extern char *tmpnam_r (char __s[L_tmpnam]) __THROW __wur;
      |                                                    ^~~~~
/usr/include/stdio.h:223:33: error: expected ‘,’ or ‘;’ before ‘__attribute_warn_unused_result__’
  223 |    __THROW __attribute_malloc__ __wur __attr_dealloc_free;
      |                                 ^~~~~
/usr/include/stdio.h:260:46: error: expected ‘,’ or ‘;’ before ‘__attribute_warn_unused_result__’
  260 |   __attribute_malloc__ __attr_dealloc_fclose __wur;
      |                                              ^~~~~
/usr/include/stdio.h: In function ‘freopen’:
/usr/include/stdio.h:267:50: error: expected declaration specifiers before ‘__attribute_warn_unused_result__’
  267 |                       FILE *__restrict __stream) __wur;
      |                                                  ^~~~~
In file included from ../include/global.h:9,
                 from ../include/config.h:581,
                 from monst.c:6:
/usr/include/stdio.h:283:14: error: storage class specified for parameter ‘fopen64’
  283 | extern FILE *fopen64 (const char *__restrict __filename,
      |              ^~~~~~~
In file included from /usr/include/features.h:486,
                 from /usr/include/x86_64-linux-gnu/bits/libc-header-start.h:33,
                 from /usr/include/stdio.h:27,
                 from ../include/global.h:9,
                 from ../include/config.h:581,
                 from monst.c:6:
/usr/include/stdio.h:285:46: error: expected ‘,’ or ‘;’ before ‘__attribute_warn_unused_result__’
  285 |   __attribute_malloc__ __attr_dealloc_fclose __wur;
      |                                              ^~~~~
/usr/include/stdio.h:288:52: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘__attribute_warn_unused_result__’
  288 |                         FILE *__restrict __stream) __wur;
      |                                                    ^~~~~
In file included from ../include/global.h:9,
                 from ../include/config.h:581,
                 from monst.c:6:
/usr/include/stdio.h:293:14: error: storage class specified for parameter ‘fdopen’
  293 | extern FILE *fdopen (int __fd, const char *__modes) __THROW
      |              ^~~~~~
In file included from /usr/include/features.h:486,
                 from /usr/include/x86_64-linux-gnu/bits/libc-header-start.h:33,
                 from /usr/include/stdio.h:27,
                 from ../include/global.h:9,
                 from ../include/config.h:581,
                 from monst.c:6:
/usr/include/stdio.h:294:46: error: expected ‘,’ or ‘;’ before ‘__attribute_warn_unused_result__’
  294 |   __attribute_malloc__ __attr_dealloc_fclose __wur;
      |                                              ^~~~~
In file included from ../include/global.h:9,
                 from ../include/config.h:581,
                 from monst.c:6:
/usr/include/stdio.h:300:14: error: storage class specified for parameter ‘fopencookie’
  300 | extern FILE *fopencookie (void *__restrict __magic_cookie,
      |              ^~~~~~~~~~~
In file included from /usr/include/features.h:486,
                 from /usr/include/x86_64-linux-gnu/bits/libc-header-start.h:33,
                 from /usr/include/stdio.h:27,
                 from ../include/global.h:9,
                 from ../include/config.h:581,
                 from monst.c:6:
/usr/include/stdio.h:303:46: error: expected ‘,’ or ‘;’ before ‘__attribute_warn_unused_result__’
  303 |   __attribute_malloc__ __attr_dealloc_fclose __wur;
      |                                              ^~~~~
In file included from ../include/global.h:9,
                 from ../include/config.h:581,
                 from monst.c:6:
/usr/include/stdio.h:308:14: error: storage class specified for parameter ‘fmemopen’
  308 | extern FILE *fmemopen (void *__s, size_t __len, const char *__modes)
      |              ^~~~~~~~
In file included from /usr/include/features.h:486,
                 from /usr/include/x86_64-linux-gnu/bits/libc-header-start.h:33,
                 from /usr/include/stdio.h:27,
                 from ../include/global.h:9,
                 from ../include/config.h:581,
                 from monst.c:6:
/usr/include/stdio.h:309:54: error: expected ‘,’ or ‘;’ before ‘__attribute_warn_unused_result__’
  309 |   __THROW __attribute_malloc__ __attr_dealloc_fclose __wur;`

**Comment by nhmall** [comment]
Near the top of sys/unix/Install.unx, there is the following note:

**NB: We are gradually replacing this process with the process documented in
the file NewInstall.unx.**

You could try using the steps documented in sys/unix/NewInstall.unx to see if they give you better results.

Also, your version of gcc and its library were released after NetHack-3.6.6 was released. You could try applying the following patch to get you past a known incompatibility with more recent versions of gcc and its library:

```
diff --git a/include/tradstdc.h b/include/tradstdc.h
index 12a0d4afe..a70698be4 100644
--- a/include/tradstdc.h
+++ b/include/tradstdc.h
@@ -429,12 +429,14 @@ typedef genericptr genericptr_t; /* (void *) or (char *) */
 #if __GNUC__ >= 3
 #define UNUSED __attribute__((unused))
 #define NORETURN __attribute__((noreturn))
+#if !defined(__linux__) || defined(GCC_URWARN)
 /* disable gcc's __attribute__((__warn_unused_result__)) since explicitly
    discarding the result by casting to (void) is not accepted as a 'use' */
 #define __warn_unused_result__ /*empty*/
 #define warn_unused_result /*empty*/
 #endif
 #endif
+#endif

 #ifndef PRINTF_F
 #define PRINTF_F(f, v)
```

 Save it to the top of the NetHack tree as nh366.diff and apply with:
```
patch -p1 -l -i nh366.diff
```
Edit src/Makefile and ensure that WINTTYLIB is set as follows prior to invoking 'make all':
`WINTTYLIB=-lncurses -ltinfo`

You'll likely encounter a lot of compiler warnings, most of which have already been fixed in the unreleased NetHack-3.7 development branch.

**Comment by nhmall** [comment]
Work-around suggested and provided for NetHack  3.6.6, and the in-development NetHack-3.7 already has the fixes.
Closing this issue.
